### PR TITLE
Cosmetically refactor delete handling

### PR DESCRIPTION
**Pull request by ironcladlou**
Some small cosmetic refactoring to the HostedCluster delete handling
to make the finalization decision-making hopefully more clear.

### CODE REVIEW

**Comment by ironcladlou** [comment]
cc @sjenning

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -491,53 +491,50 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
...
 
 	for key := range nodePools {
 		if err := r.Delete(ctx, &nodePools[key]); err != nil && !apierrors.IsNotFound(err) {
-			return result, fmt.Errorf("failed to delete nodePool %q for cluster %q: %w", nodePools[key].GetName(), req.Name, err)
+			return false, fmt.Errorf("failed to delete nodePool %q for cluster %q: %w", nodePools[key].GetName(), hc.Name, err)
 		}
 	}
 
-	r.Log.Info("Deleting Cluster", "clusterName", req.Name, "clusterNamespace", targetNamespace)
+	r.Log.Info("Deleting Cluster", "clusterName", hc.Name, "clusterNamespace", hc.Namespace)
```
Comment (sjenning) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:509]: `targetNamespace` is `hc.Name` not `hc.Namespace` right?

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 2]
```diff
@@ -491,53 +491,50 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
...
 	cluster := &capiv1.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: targetNamespace,
+			Name:      hc.Name,
+			Namespace: hc.Namespace,
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:513]: this needs to be the targetNamespace

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 3]
```diff
@@ -491,53 +491,50 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
...
-			RequeueAfter: time.Duration(5 * time.Second),
-		}
-		r.Log.Info("Waiting for Cluster deletion", "clusterName", req.Name, "clusterNamespace", targetNamespace)
-		return result, nil
+		r.Log.Info("waiting for Cluster deletion", "clusterName", hc.Name, "clusterNamespace", hc.Namespace)
+		return false, nil
 	}
 
-	r.Log.Info("Deleting target namespace", "namespace", targetNamespace)
+	r.Log.Info("Deleting target namespace", "namespace", hc.Namespace)
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:527]: hc.Namescape is not the target namespace.
hc.GetName() is, therfore why we have `targetNamespace = req.Name`

**Comment by sjenning** [comment]
superseded by https://github.com/openshift/hypershift/pull/45

**Pull request closed without merging**